electron_app/DownloadManager.js

```diff
@@ -1,25 +1,48 @@
 const Downloader = require('./Downloader');
 const path = require('path');
-const { handleFolder } = require('./helpers/handleDisk');
+const { handleFolder, fileAlreadyExists } = require('./helpers/handleDisk');
 const downloadStats = require('./helpers/downloadInfo').downloadStats;
 const handleEvents = require('./helpers/handleEvents');
 
 let filePath = '';
 let allDownloads = [];
+let fileExists = true;
 
-const downloadManager = (items, access_token, win) => {
+const downloadManager = (items, access_token, electronWin) => {
   items.forEach(item => {
-    if (item.preserveStructure) {
-      const structurePath = item.path.substring(item.path.lastIndexOf('/'), 0);
-      handleFolder(path.join(item.destination, structurePath)).then(result => {
-        allDownloads.push(processDownload(access_token, item, result, win));
-      });
-    } else {
-      allDownloads.push(processDownload(access_token, item, item.destination, win));
-    }
+    let count = 0;
+    let fileName = item.name;
+    do {
+      fileExistsInDestination(item, fileName);
+      if (fileExists) {
+        count++;
+        fileName = item.name.substring(0, item.name.lastIndexOf('.')) + '(' + (count) + ')' +
+          item.name.substring(item.name.lastIndexOf('.'));
+      } else {
+        item.name = fileName;
+        processItem(item, access_token, electronWin);
+      }
+    } while (fileExists);
   });
 };
 
+const fileExistsInDestination = (item, nameCheck) => {
+  fileExists = item.preserveStructure ?
+    fileAlreadyExists(path.join(item.destination + '/' + item.path.substring(item.path.lastIndexOf('/'), 0) + '/' + nameCheck))
```

Comment:
I'm pretty sure this is not the right way to use `path.join()`; you want to avoid hardcoding `/` and pass in each path element as a separate argument, e.g. `path.join(item.destination, item.path.substring(...), nameCheck)`

Reply:
Fixed!